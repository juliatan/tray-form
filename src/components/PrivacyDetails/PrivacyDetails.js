/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { Formik, Field, Form } from 'formik';
import classes from './PrivacyDetails.module.scss';

const privacyDetails = (props) => {
  return (
    <div className={classes.PrivacyDetails}>
      <Formik
        initialValues={{ newsletters: [] }}
        onSubmit={(values, { setSubmitting }) => {
          props.saveDetails(values);
          setSubmitting(false);
        }}
      >
        {({ isSubmitting }) => (
          <Form>
            {/* <div role="group" aria-labelledby="checkbox-group"> */}
            <label>
              <Field type="checkbox" name="newsletters" value="trayProduct" />
              Receive updates about Tray.io product by email
            </label>
            <label>
              <Field type="checkbox" name="newsletters" value="otherProducts" />
              Receive communication by email for other products created by the
              Tray.io team
            </label>
            {/* </div> */}

            <button
              type="submit"
              disabled={isSubmitting}
              className={classes.submitButton}
            >
              Submit
            </button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default privacyDetails;
