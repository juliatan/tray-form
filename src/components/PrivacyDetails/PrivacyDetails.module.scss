@import '../../assets/stylesheets/colours';

.PrivacyDetails {
  form {
    display: flex;
    flex-direction: column;
  }

  input {
    margin: 8px 8px 8px 0;
  }
}
