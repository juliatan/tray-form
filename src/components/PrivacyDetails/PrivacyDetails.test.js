import React from 'react';
import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import PrivacyDetails from './PrivacyDetails';

describe('PrivacyDetails component', () => {
  let saveDetails;
  let component;

  beforeEach(() => {
    saveDetails = jest.fn();
    component = <PrivacyDetails saveDetails={saveDetails} />;
  });

  afterEach(cleanup);

  it('should render', () => {
    const { asFragment } = render(component);
    expect(asFragment(component)).toMatchSnapshot();
  });

  it('should submit form when both options checked', async () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId('trayProduct'));
    fireEvent.click(getByTestId('otherProducts'));
    fireEvent.click(getByTestId('privacyDetailsSubmit'));

    await waitFor(() =>
      expect(saveDetails).toHaveBeenCalledWith({
        newsletters: ['trayProduct', 'otherProducts'],
      }),
    );
  });

  it('should submit form when trayProduct is checked', async () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId('trayProduct'));
    fireEvent.click(getByTestId('privacyDetailsSubmit'));

    await waitFor(() =>
      expect(saveDetails).toHaveBeenCalledWith({
        newsletters: ['trayProduct'],
      }),
    );
  });

  it('should submit form when otherProducts is checked', async () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId('otherProducts'));
    fireEvent.click(getByTestId('privacyDetailsSubmit'));

    await waitFor(() =>
      expect(saveDetails).toHaveBeenCalledWith({
        newsletters: ['otherProducts'],
      }),
    );
  });

  it('should submit form when no options are checked', async () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId('privacyDetailsSubmit'));

    await waitFor(() =>
      expect(saveDetails).toHaveBeenCalledWith({
        newsletters: [],
      }),
    );
  });
});
