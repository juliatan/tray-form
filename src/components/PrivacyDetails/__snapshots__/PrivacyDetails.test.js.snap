// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyDetails component should render 1`] = `
<DocumentFragment>
  <div
    class="PrivacyDetails"
    data-testid="privacyDetails"
  >
    <form
      action="#"
    >
      <label>
        <input
          data-testid="trayProduct"
          name="newsletters"
          type="checkbox"
          value="trayProduct"
        />
        Receive updates about Tray.io product by email
      </label>
      <label>
        <input
          data-testid="otherProducts"
          name="newsletters"
          type="checkbox"
          value="otherProducts"
        />
        Receive communication by email for other products created by the Tray.io team
      </label>
      <button
        class="Button"
        data-testid="privacyDetailsSubmit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
